src/modules.ts

```diff
@@ -1,24 +1,34 @@
-import { resolve } from 'path'
+import { dirname, resolve } from 'path'
 import { promises as fs } from 'fs'
+import { fileURLToPath } from 'url'
 import type { BuildResult } from 'workbox-build'
 import type { ResolvedConfig } from 'vite'
 import type { ResolvedVitePWAOptions } from './types'
 import { logWorkboxResult } from './log'
 import { defaultInjectManifestVitePlugins } from './constants'
 
-function loadWorkboxBuild() {
+const _dirname = typeof __dirname !== 'undefined'
+  ? __dirname
+  : dirname(fileURLToPath(import.meta.url))
+
+async function loadWorkboxBuild() {
   // Uses require to lazy load.
   // "workbox-build" is very large and it makes config loading slow.
   // Since it is not always used, load this when it is needed.
 
-  return require('workbox-build')
+  try {
+    return await import('workbox-build')
```

Comment:
We we need interop `default` here?

Reply:
I will check it again but it works, I can split it to check for default

Reply:
done